apps/docs/pages/guides/auth/auth-helpers/nextjs-server-components.mdx

```diff
@@ -164,6 +166,8 @@ export const config = {
 
 <TabPanel id="ts" label="TypeScript">
 
+Create a new file at `/pages/middleware.ts` and populate with the following:
```

Comment:
I wasn't sure where to place the middleware file, but got it to work by placing it in the _pages_ directory. Is this the intended location for a middleware?

Reply:
Thanks @danherrera — not anymore. As of 12.2 I think, it's recommended to be at the root — on the same level as /pages/